src/main/java/com/powsybl/caseserver/service/S3CaseService.java

```diff
@@ -0,0 +1,558 @@
+        Path tempCasePath;
+        try {
+            FileAttribute<Set<PosixFilePermission>> attr = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------"));
+            tempdirPath = Files.createTempDirectory(caseUuid.toString(), attr);
+
+            // Create parent directory if necessary
+            Path parentPath = Paths.get(filename).getParent();
+            if (parentPath != null) {
+                Files.createDirectory(tempdirPath.resolve(parentPath), attr);
+            }
+            // after this line, need to cleanup the dir
+        } catch (IOException e) {
+            throw CaseException.createTempDirectory(caseUuid, e);
+        }
+        try {
+            tempCasePath = tempdirPath.resolve(filename);
+            try {
+                contentInitializer.accept(tempCasePath);
+            } catch (CaseException e) {
+                throw CaseException.initTempFile(caseUuid, e);
+            } catch (Throwable ex) {
+                throw CaseException.initTempFile(caseUuid, ex);
+            }
+            // after this line, need to cleanup the file
+            try {
+                try {
+                    return f.apply(tempCasePath);
+                } catch (CaseException e) {
+                    throw CaseException.createFileNotImportable(tempdirPath);
+                } catch (Throwable t) {
+                    throw CaseException.processTempFile(caseUuid, t);
+                }
+            } finally {
+                try {
+                    Files.delete(tempCasePath);
+                } catch (IOException e) {
+                    LOGGER.error("Error cleaning up temporary case file", e);
+                }
+            }
+        } finally {
+            try {
+                if (Files.exists(tempdirPath)) {
+                    FileUtils.deleteDirectory(tempdirPath.toFile());
+                }
+            } catch (IOException e) {
+                LOGGER.error("Error cleaning up temporary case directory: " + tempdirPath, e);
+            }
+        }
+    }
+
+    // downloads from s3 and cleanup
+    public <R, T extends Throwable> R withS3DownloadedTempPath(UUID caseUuid, FailableFunction<Path, R, T> f) {
+        return withS3DownloadedTempPath(caseUuid, null, f);
+    }
+
+    public <R, T extends Throwable> R withS3DownloadedTempPath(UUID caseUuid, String caseFileKeyFromCaller, FailableFunction<Path, R, T> f) {
+        String caseFileKey = Objects.requireNonNullElse(caseFileKeyFromCaller, getCaseFileObjectKey(caseUuid));
+        String filename = parseFilenameFromKey(caseFileKey);
+        return withTempCopy(caseUuid, filename, path ->
+                        s3Client.getObject(GetObjectRequest.builder().bucket(bucketName).key(caseFileKey).build(), path),
+                f);
+    }
+
+    @Override
+    public String getFormat(UUID caseUuid) {
+        String format = getCaseMetadata(caseUuid).get(FORMAT_HEADER_KEY);
+        if (format == null) {
```

Comment:
I don't think we want this code. it will work with metadata

Reply:
Done.

Reply:
Done